### PR TITLE
fix: show error in edit text when email is invalid.

### LINKED ISSUE #927: No error message when user clicks on login button with email and password fields empty.
**Describe the bug**

In the login fragment when the user clicks on login button without entering email and password fields nothing happens, I think there should be error message like "email and password fields are required for login". I think this error message is required because it will enhance user experience.

**To Reproduce**
1. Go to loginFragment
2. Click on login button without entering email, password fields.
3. No error message pop's up, the UI remains same.

**Expected behavior**

A snackbar with the message "email and password fields required to login" or error message with red icon in the email, password edit text.

**Screenshots**

ezgif com-video-to-gif-1

**Smartphone Info:**

|               | |
|---------------|-|
|Device         | [Redmi note 5] |
|Android Version| [Oreo 8.1]  |

**Would you like to work on the issue?**

- [x] Yes

### PR DESCRIPTION
**Pull request by Cool-fire**
Fixes #927 

Changes: added snackbar when user clicks on login button with wrong input in email field and also a snackbar when user clicks on login button with both email and password fields empty. previously for both these cases the login button is disabled so nothing happens when clicked on button.

Screenshots for the change:
1. first screenshot is when email input is wrong and user clicks on login button
2. second screenshot is when both fields are empty and user clicks on login button
3. third screenshot is when user typed his password but not email

### CODE REVIEW

**Comment by Cool-fire** [comment]
@liveHarshit, is there anything wrong with this issue, so that I would know if anything can be improved in this pull request.

**Comment by liveHarshit** [comment]
> @liveHarshit, is there anything wrong with this issue, so that I would know if anything can be improved in this pull request.

Actually, IMO if the button is disabled, and not clickable user can see that fields are blanks. Why show an error message which is not required?.

**Comment by iamareebjamal** [comment]
If you want to show that, use TextInputLayout error fields

**Comment by Cool-fire** [comment]
@iamareebjamal will use the TextInputlayout error fields, and will update the pull request after necessary changes.

**Comment by Cool-fire** [comment]
@iamareebjamal @liveHarshit please review the changes.

**Comment by iamareebjamal** [comment]
Squash your 26 commits into 1

File: app/src/main/res/layout/fragment_login.xml [hunk 1]
```diff
@@ -76,7 +76,7 @@
                     android:id="@+id/loginButton"
                     android:layout_width="match_parent"
                     android:layout_height="wrap_content"
-                    android:enabled="false"
+                    android:enabled="true"
```
Comment (iamareebjamal) [inline app/src/main/res/layout/fragment_login.xml]: This should only enable when everything is correct

Comment (Cool-fire) [reply app/src/main/res/layout/fragment_login.xml]: Ok @iamareebjamal sir, will do that and will update the pull request with necessary changes.

Comment (Cool-fire) [reply app/src/main/res/layout/fragment_login.xml]: Thanks @iamareebjamal will use textwatcher and will update the pull request with necessary changes.

**Comment by Cool-fire** [comment]
@iamareebjamal I tried to squash all my commits to one by doing git rebase -i HEAD~[N] but I think it didn't worked out. And before pushing I did git pull upstream development --rebase. Could you please help me with this.

**Comment by Cool-fire** [comment]
@iamareebjamal  please review the changes.

**Comment by iamareebjamal** [comment]
50 commits and 11 file changes. I am sure I won't be able to look at this unless there are at most 1 commit and 3 file changes. At most

**Pull request closed without merging**